src/parser.rs

```diff
@@ -149,66 +149,20 @@ impl<'a> Parser<'a> {
 
     fn parse_expression(&mut self) -> Option<Expression> {
         let start = self.position;
+
         if let Some(subexpression) = self.parse_subexpression() {
+            let boxed = Box::new(subexpression);
```

Comment:
Can we keep the name `subexpression`, that feels more useful than boxed?

Reply:
I will fix in a later refactoring PR